python/rmm/tests/test_rmm.py

```diff
@@ -586,6 +586,34 @@ def test_cuda_async_memory_resource_threshold(nelem, alloc):
     array_tester("u1", 2 * nelem, alloc)  # should trigger release
 
 
+@pytest.mark.parametrize(
+    "mr", [rmm.mr.CudaMemoryResource, rmm.mr.CudaAsyncMemoryResource]
+)
+def test_limiting_resource_adaptor(mr):
+    if not _CUDAMALLOC_ASYNC_SUPPORTED:
+        pytest.skip("cudaMallocAsync not supported")
+
+    cuda_mr = mr()
+
+    allocation_limit = 1 << 20
+    num_buffers = 2
+    buffer_size = allocation_limit / num_buffers
+
+    mr = rmm.mr.LimitingResourceAdaptor(
+        cuda_mr, allocation_limit=allocation_limit
+    )
+    assert mr.get_allocation_limit() == allocation_limit
+
+    rmm.mr.set_current_device_resource(mr)
+
+    buffers = [rmm.DeviceBuffer(size=buffer_size) for _ in range(num_buffers)]
+
+    assert mr.get_allocated_bytes() == buffer_size * num_buffers
+
+    with pytest.raises(MemoryError):
+        buffers.append(rmm.DeviceBuffer(size=buffer_size))
```

Comment:
Yes, it will also fail. The same `buffer_size` was used because it seemed more straightforward to use the same size. Any strong opinions on which to use here?

Reply:
Let's make a small allocation here, so:

```suggestion
        rmm.DeviceBuffer(size=1)
```

Reply:
Thanks, had to also add https://github.com/rapidsai/rmm/pull/1327/commits/e37c70f5af339e70f50322e43d262afb085acbd3 to make sure linter is happy that we use `buffers` somewhere, plus it makes more sense to check the actual allocation sizes.